plugin/vcs/vcs.go

```diff
@@ -96,27 +102,30 @@ type RepositoryMember struct {
 type Provider interface {
 	// Returns the API URL for a given VCS instance URL
 	APIURL(instanceURL string) string
-
 	// Try to use this provider as an auth provider and fetch the user info from the OAuth context
 	//
 	// oauthCtx: OAuth context to write the file content
 	// instanceURL: VCS instance URL
 	TryLogin(ctx context.Context, oauthCtx common.OauthContext, instanceURL string) (*UserInfo, error)
-
 	// Fetch the user info of the given userID
 	//
 	// oauthCtx: OAuth context to write the file content
 	// instanceURL: VCS instance URL
 	// userID: the ID of the desired user
 	FetchUserInfo(ctx context.Context, oauthCtx common.OauthContext, instanceURL string, userID int) (*UserInfo, error)
-
 	// Fetch all active members of a given repository
 	//
 	// oauthCtx: OAuth context to write the file content
 	// instanceURL: VCS instance URL
 	// repositoryID: the repository ID from the external VCS system (note this is NOT the ID of Bytebase's own repository resource)
 	FetchRepositoryActiveMemberList(ctx context.Context, oauthCtx common.OauthContext, instanceURL string, repositoryID string) ([]*RepositoryMember, error)
-
+	// Fetch the tree nodes of a repository
+	//
+	// oauthCtx: OAuth context to read the repo tree
+	// instanceURL: VCS instance URL
+	// repositoryID: the repository ID from the external VCS system (note this is NOT the ID of Bytebase's own repository resource)
+	// ref: the unique name of a repo tree, could be a branch name in GitLab or a tree sha in GitHub.
+	FetchRepoNodeList(ctx context.Context, oauthCtx common.OauthContext, instanceURL string, repositoryID string, ref string) ([]*RepoTreeNode, error)
```

Comment:
Do need to supply filePath? Looks like the current implementation always fetches the root directory recursively. And for the sheet integration, we likely will fetch under a folder say bytebase/scripts/

Reply:
It's nice to have a path param, but GitHub API don't have it: https://docs.github.com/en/rest/reference/git#get-a-tree. 

So we can use `query` payload instead of `ref` in func to eliminate the difference: 
```
FetchRepoNodeList(ctx context.Context, oauthCtx common.OauthContext, instanceURL string, repositoryID string, **query string**) ([]*RepoTreeNode, error)
```

wdyt

Reply:
For FetchRepoNodeList, it's better to hide the difference of the underlying VCS API behavior detail.

So, if GitHub only returns all files under the repository, then it's the FetchRepoNodeList's work to filter out the unneeded files

Reply:
> FetchRepoNodeList -> FetchRepositoryTree

Because we want the node(file/folder) list not a tree, so I prefer to name it with `FetchRepositoryNodeList`.

Reply:
Maybe FetchRepositoryFileList?

Because Node usually indicates it has references to its parent/child. And according to our current usage, we just want a flattened list.